src/tox_current_env/hooks4.py

```diff
@@ -0,0 +1,226 @@
+from tox.plugin import impl
+from tox.tox_env.python.api import PythonInfo
+from tox.tox_env.python.runner import PythonRun
+
+
+@impl
+def tox_add_option(parser):
+    parser.add_argument(
+        "--current-env",
+        action="store_true",
+        dest="current_env",
+        default=False,
+        help="Run tests in current environment, not creating any virtual environment",
+    )
+    parser.add_argument(
+        "--print-deps-only",
+        action="store_true",
+        dest="print_deps_only",
+        default=False,
+        help="Deprecated, equivalent to `--print-deps-to -`",
+    )
+    parser.add_argument(
+        "--print-deps-to",
+        "--print-deps-to-file",
+        action="store",
+        type=argparse.FileType("w"),
+        metavar="FILE",
+        default=False,
+        help="Don't run tests, only print the dependencies to the given file "
+        + "(use `-` for stdout)",
+    )
+    parser.add_argument(
+        "--print-extras-to",
+        "--print-extras-to-file",
+        action="store",
+        type=argparse.FileType("w"),
+        metavar="FILE",
+        default=False,
+        help="Don't run tests, only print the  names of the required extras to the given file "
+        + "(use `-` for stdout)",
+    )
+
+
+@impl
+def tox_register_tox_env(register):
+    register.add_run_env(CurrentEnv)
+    register.add_run_env(PrintEnv)
+
+
+@impl
+def tox_add_core_config(core_conf, config):
+    if config.options.current_env:
+        config.options.default_runner = "current-env"
+        return
+
+    if config.options.print_deps_only:
+        warnings.warn(
+            "--print-deps-only is deprecated; use `--print-deps-to -`",
+            DeprecationWarning,
+        )
+        if not config.options.print_deps_to:
+            config.options.print_deps_to = sys.stdout
+        else:
+            raise RuntimeError(
+                "--print-deps-only cannot be used together " "with --print-deps-to"
+            )
+
+    if getattr(config.options.print_deps_to, "name", object()) == getattr(
+        config.options.print_extras_to, "name", object()
+    ):
+        raise RuntimeError(
+            "The paths given to --print-deps-to and --print-extras-to cannot be identical."
+        )
+
+    if config.options.print_deps_to or config.options.print_extras_to:
+        config.options.default_runner = "print-env"
+        return
+
+    # No options used - switch back to the standard runner
+    # Workaround for: https://github.com/tox-dev/tox/issues/2264
+    config.options.default_runner = "virtualenv"
+
+
+class CurrentEnv(PythonRun):
+    def __init__(self, create_args):
+        self._executor = None
+        self._installer = None
+        self._path = []
+        super().__init__(create_args)
+
+    @staticmethod
+    def id():
+        return "current-env"
+
+    @property
+    def _default_package_tox_env_type(self):
+        return None
```

Comment:
Eh, why? You still need to handle packaging even if you're running in current env. Or else how is packaging handled?

Reply:
All packages are already installed before tox is executed - their installation is handled on RPM level. We are using this only to run all the commands from the tox configuration with system-wide-installed packages.

Reply:
That's fine then. Though perhaps there should be some type check package is installed or not, but I assume that's implicitly assumed correct.

Reply:
Yeah, see the README. When using the current env, you are responsible for installing the packages.